git-worktree/src/lib.rs

```diff
@@ -1 +1,126 @@
 #![forbid(unsafe_code, rust_2018_idioms)]
+//! Git Worktree
+
+use git_hash::oid;
+use git_object::bstr::ByteSlice;
+use quick_error::quick_error;
+use std::convert::TryFrom;
+use std::fs;
+use std::fs::create_dir_all;
+use std::path::Path;
+use std::time::Duration;
+
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+
+quick_error! {
+    #[derive(Debug)]
+    pub enum Error {
+        Utf8Error(err: git_object::bstr::Utf8Error) {
+            from()
+            display("Could not convert path to UTF8: {}", err)
+        }
+        TimeError(err: std::time::SystemTimeError) {
+            from()
+            display("Could not read file time in proper format: {}", err)
+        }
+        ToU32Error(err: std::num::TryFromIntError) {
+            from()
+            display("Could not convert seconds to u32: {}", err)
+        }
+        IoError(err: std::io::Error) {
+            from()
+            display("IO error while writing blob or reading file metadata or changing filetype: {}", err)
+        }
+        FindOidError(oid: git_hash::ObjectId, path: std::path::PathBuf) {
+            display("unable to read sha1 file of {} ({})", path.display(), oid.to_hex())
```

Comment:
There is no such thing as a `SHA1 file`, for all you know an object couldn't be found. This text should be reworded to reflect that.
I like that the path at which the object would be checked out at is mentioned as well.

Reply:
The same error is used in the git source code you linked: https://github.com/git/git/blob/main/entry.c#L297:L299

Reply:
Done: https://github.com/Byron/gitoxide/pull/315/commits/a4b880cf17665b61e3f7f193de57704b1db5318f